presto-main/src/main/java/com/facebook/presto/operator/scalar/ScalarFunctionImplementation.java

```diff
@@ -34,6 +34,10 @@
     private final List<ArgumentProperty> argumentProperties;
     private final MethodHandle methodHandle;
     private final Optional<MethodHandle> instanceFactory;
+
+
+    private final List<ScalarImplementationChoice> choices;
```

Comment:
It appears like the changes you made to this class is half way. I'll skip this class for now.